pkg/storkctl/clusterpair.go

```diff
@@ -522,100 +449,189 @@ func newCreateClusterPairCommand(cmdFactory Factory, ioStreams genericclioptions
+						Name:      clusterPairName,
+						Namespace: cmdFactory.GetNamespace(),
+						Annotations: map[string]string{
+							skipResourceAnnotation: "true",
+							storkCreatedAnnotation: "true",
+						},
+					},
+					Data: credentialData,
+					Type: v1.SecretTypeOpaque,
+				}
+				_, err = core.Instance().CreateSecret(secret)
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
 
-			if len(srcToken) == 0 {
-				pxAuthTokenSrc, pxAuthSecretNamespaceSrc, err = getPXAuthToken(sFile)
+				backupLocation.Location.SecretConfig = clusterPairName
+				var annotations = make(map[string]string)
+				annotations[skipResourceAnnotation] = "true"
+				_, err = storkops.Instance().CreateBackupLocation(backupLocation)
 				if err != nil {
-					printMsg(fmt.Sprintf("Got error while fetching px auth token in source cluster: %v", err), ioStreams.Out)
+					err := fmt.Errorf("unable to create ObjectstoreLocation in source. Err: %v", err)
+					util.CheckErr(err)
+					return
 				}
-				if len(pxAuthTokenSrc) > 0 {
-					printMsg("Fetching px token with auth token in source cluster", ioStreams.Out)
+				printMsg(fmt.Sprintf("ObjectstoreLocation %v created on source cluster in namespace %v\n", clusterPairName, cmdFactory.GetNamespace()), ioStreams.Out)
+
+				storkops.Instance().SetConfig(dconf)
+				core.Instance().SetConfig(dconf)
+				printMsg(fmt.Sprintf("Creating Secret and ObjectstoreLocation in destination cluster in namespace %v...", cmdFactory.GetNamespace()), ioStreams.Out)
+				_, err = core.Instance().CreateSecret(secret)
+				if err != nil {
+					util.CheckErr(err)
+					return
 				}
-				token, err := getPXToken(sEP, pxAuthTokenSrc)
+				_, err = storkops.Instance().CreateBackupLocation(backupLocation)
 				if err != nil {
-					err := fmt.Errorf("got error while fetching px token in source cluster %s. Err: %v", sEP, err)
+					err := fmt.Errorf("unable to create ObjectstoreLocation in destination. Err: %v", err)
 					util.CheckErr(err)
 					return
 				}
-				srcToken = token
+				printMsg(fmt.Sprintf("ObjectstoreLocation %v created on destination cluster in namespace %v\n", clusterPairName, cmdFactory.GetNamespace()), ioStreams.Out)
 			}
-			destClusterPair, err := generateClusterPair(clusterPairName, cmdFactory.GetNamespace(), sIP, sPort, srcToken, sFile, projectMappingsStr, pxAuthSecretNamespaceSrc, true, false)
+
+			printMsg("Creating a cluster pair. Direction: Source -> Destination", ioStreams.Out)
+			srcClusterPair, err := generateClusterPair(clusterPairName, cmdFactory.GetNamespace(), dIP, dPort, destToken, dFile, projectMappingsStr, pxAuthSecretNamespaceDest, false, false)
 			if err != nil {
 				util.CheckErr(err)
 				return
 			}
-			// Create cluster-pair on dest cluster
-			conf, err = getConfig(dFile).ClientConfig()
+			storkops.Instance().SetConfig(sconf)
+			core.Instance().SetConfig(sconf)
+			srcClusterPair.Spec.Options[storkv1.BackupLocationResourceName] = finalBackupLocationName
```

Comment:
You are using `finalBackupLocationName` interchangeably which is confusing. In certain cases you set the Options to backupLocation.Name and sometimes you use this variable.

Reply:
I will remove this variable. Will modify the code, to overwrite the backuplocationName only in case of it is empty.